### ISSUE TITLE
HTML5 Input type=number broken in Google Chrome

### ISSUE THREAD

**Issue by nvaccessAuto** [nvaccess/nvda#5422]
**Reported by spanchang02 on 2015-10-16 20:21**
NVDA 2015.3 does not work with HTML5 input type=number in Google Chrome 46.
It does announce it as a spin box however.
By using up/down arrow keys with NVDA 2015.3 one could increment/decrement values in   this  type of control in Google Chrome 45.
Example: The Age and Distance Walked fields on
www.mindoversight.com/demo/Html5Form.htm 

Tested on Windows 8.1

**Comment by jcsteh** [comment]
@spanchang [noted elsewhere that this issue appeared to have been resolved](https://github.com/nvaccess/nvda/issues/5939#issuecomment-217941155). That may have been true, but these controls appear to be broken again now.

**Comment by jcsteh** [comment]
STR:

1. Open this URL in Chrome: data:text/html,<input type="number" value="1">
2. In browse mode, press enter on the spin button to switch to focus mode.
3. Press up and down arrows to increase and decrease the value.
 * Expected: The new value should be read on each change.
 * Actual: "blank"

The text interface on this control is somewhat broken:

1. There are two embedded objects: an editable section containing the value and a spin button containing two buttons.
2. The caret offset points at the child spin button (instead of the section) when the caret is on the second digit of a two digit number.
3. When the caret is at the end of the number (as it is when you press up/down arrow), retrieving the line offsets never includes the editable section (and thus never includes the number).

I'll file an issue against Chrome.

**Comment by jcsteh** [comment]
Filed: https://bugs.chromium.org/p/chromium/issues/detail?id=730965

Note that the implementation has changed once again, though with similar results for NVDA users. See the above Chromium issue for details.

**Comment by ehollig** [comment]
It appears that [Issue 730965](https://bugs.chromium.org/p/chromium/issues/detail?id=730965) has been fixed in Chrome.
Performing the steps provided by @jcsteh in https://github.com/nvaccess/nvda/issues/5422#issuecomment-287688470 the value of the slider is read both in browse and focus modes. Closing

**Comment by leonardder** [comment]
interesting enough, I can still reproduce this in both Edge and Chrome. The value reads in browse mode, but incrementing/decrementing in focus mode does not read.

**Comment by woody-li** [comment]
Able to reproduce in focus mode at Chrome 92.  Tried two cases below:

`data:text/html, <input type="number" value=2 />` 
* Browse mode & focus mode: _spin button  editable    2_

`data:text/html, <span>Spin:</span><input type="number" value=2 />`
* Browse mode: _Spin:  spin button  editable    2_
* Focus mode: _spin button  editable  blank_

**Comment by Adriani90** [comment]
cc: @ObjectInSpace, @alevelthal

**Comment by aleventhal** [comment]
Thanks @Adriani90 . Is there an action item for Chrome? 
Do you know if the content works in JAWS?
Also would not be wrong to try in a recent unstable version of Chrome, e.g. Chrome Canary (95).

**Comment by jcsteh** [comment]
This is broken because on input type="number", Chrome (including Canary)
exposes the editable state and the IAccessibleText interface, but doesn't
expose any text via IAccessibleText (nCharacters is 0). It does expose an
accValue, but this gets ignored if the control is editable text.

**Comment by sevapopov2** [comment]
This issue appears also in NVDA 2021.2 beta1 and doesn't appear with Jaws. I have checked it today and everything works fine with JFW.

**Comment by aleventhal** [comment]
Work in progress here: https://chromium-review.googlesource.com/c/chromium/src/+/3115245
I expect that to land in Chrome 95, possibly in 94.

**Comment by aleventhal** [comment]
This should be fixed in tomorrow's Canary, version 95.0.4622.0 or later.

Just a note: input type=number gets the spinner role, because the user can press up/down.
There are some other spinner roles used inside of date, datetime, datetime-local, month, time, and week input types, e.g. ```<input type=week>```. If anyone wants to test those with NVDA as well, that would be helpful.

**Comment by woody-li** [comment]
> Able to reproduce in focus mode at Chrome 92. Tried two cases below:
> 
> `data:text/html, <input type="number" value=2 />`
> 
> * Browse mode & focus mode: _spin button  editable    2_
> 
> `data:text/html, <span>Spin:</span><input type="number" value=2 />`
> 
> * Browse mode: _Spin:  spin button  editable    2_
> * Focus mode: _spin button  editable  blank_

Checked in Chrome 95.0.4628.0 canary with the cases above.
Browse mode and focus mode both reads the number value.

**Comment by Adriani90** [comment]
Closing as works for me in Canari 95. For other datetime inputs there is a separate issue in place.
